File: backend/flutter-movie-backend-main/utils/response.js
class Response{
    constructor(status, message, data){
        this.status = status;
        this.message = message;
        this.data = data;
    }

    static success(data){
        return new Response(200, null, data);
    }
    static error(message){
        return new Response(400, message, null);
    }
    static unauthorized(message){
        return new Response(401, message, null);
    }
}

module.exports = Response;

File: flutter/flutter-movie-main/lib/models/response.dart
class Response{
  final int status;
  final String? message;
  final dynamic data;

  Response({required this.status, required this.message, required this.data});

  static Response fromJson(responseBody) {
    return Response(
      status: responseBody['status'],
      message: responseBody['message'],
      data: responseBody['data'],
    );
  }
}